General Tools
=============

* [Samtools](https://github.com/samtools) --- This is a general workhorse genomics tools.  It's actually an entire suite 
  of related tools for processing of SAM, BAM and CRAM format files.  Moreover, it is the de-facto implementation of 
  SAM and BAM parsing and manipulation in C, and provides and API for interacting with these types of files from C/C++.
    + [PySAM](http://pysam.readthedocs.org/en/latest/api.html) --- a Python interface to much of the Samtools API.
  
* [Bedtools](http://bedtools.readthedocs.org/en/latest/) --- This is also a standard suite of tools for processing many formats 
  of genomic files.  If you need to do something involving genomic intervals, this is the tool to look to first.

* [HTSeq](http://www-huber.embl.de/HTSeq/doc/overview.html) --- Another Python tool (and API) for interacting with      
  high-througput sequencing data.

* [Genome Assembly & Analysis Tool Box (GATB)](http://gatb.inria.fr/tutorials/) --- a very nice C++ library for        general     tasks such as k-mer counting and de Bruijn graph construction and traversal.

Aligners
--------

* [Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml) --- Popular aligner for non-spliced alignments.

* [BWA](https://github.com/lh3/bwa) --- Another popular aligner for non-spliced alignments.

* [HISAT](http://ccb.jhu.edu/software/hisat/index.shtml) --- Tool for spliced alignment (suitable for aligning RNA-seq reads      with which to build splice graphs).

* [STAR](https://github.com/alexdobin/STAR) --- Another tool for spliced alignment, STAR and HISAT are the fastest among the 
  spliced aligners, with HISAT being the newer and arguable faster of the two.

Splice Graph Tools
------------------

* [Splice Grapher](http://splicegrapher.sourceforge.net/userguide.html) --- A tool to construct splice graphs from alignments.

* [CEM](http://alumni.cs.ucr.edu/~liw/cem.html) --- Another splice graph construction tool.

Transcript Assemblers
---------------------

* [Cufflinks](http://cole-trapnell-lab.github.io/cufflinks/) --- The first and most widely-used of the genome-guided transcript
  assemblers.

* [StringTie](http://ccb.jhu.edu/software/stringtie/) --- New much faster and more accurate genome-guided assembler.

* [Bayesembler](https://github.com/bioinformatics-centre/bayesembler) --- A novel technique based on a model for simultaneous 
  transcript assembly and abundance estimation.  Currently only works with spliced-alignments generated by   
  [TopHat2](http://ccb.jhu.edu/software/tophat/index.shtml).

* [Trinity](http://trinityrnaseq.github.io/) --- One of the most common *de novo* transcriptome assemblers.

* [TransAbyss](http://www.bcgsc.ca/platform/bioinfo/software/trans-abyss) --- Another populer *de novo* transcriptome assembler.

* [SOAP De Novo Trans](http://soap.genomics.org.cn/SOAPdenovo-Trans.html) --- Yet another popular *de novo* transcriptome 
  assembler.

Other Assembly Packages
-----------------------

* [SGA](https://github.com/jts/sga) --- The String Graph Assembler, which provides an implementation for constructing 
  the string graph from a set of reads.

* [Light String Graph](https://github.com/AlgoLab/LightStringGraph) --- A *lightweight* string graph assembler.  By making
  use of external memory algorithms, this string graph assembler is able to scale to huge datasets --- worth building off of.

